File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if(list1 == null){
            return list2;
        }
        
        if(list2 == null){
            return list1;
        }
        
        ListNode head = new ListNode(0);
        ListNode tempHead = head;
        ListNode temp1 = list1;
        ListNode temp2 = list2;
        
        while(temp1 != null && temp2 != null){
            // System.out.println(temp1.val);
            if(temp1.val < temp2.val){
                tempHead.next = temp1;
                temp1 = temp1.next;
            }else{
                tempHead.next = temp2;
                temp2 = temp2.next;
            }
            tempHead = tempHead.next;
        }
        
        if(temp1 != null){
            tempHead.next = temp1;
        }
        
        if(temp2 != null){
            tempHead.next = temp2;
        }
        
        return head.next;
        
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        if(list1 == None):
            return list2
        if(list2 == None):
            return list1
        
        head = None
        temp1 = list1
        temp2 = list2
        
        if(list1.val < list2.val):
            head = list1
            temp1 = temp1.next
        else:
            head = list2
            temp2 = temp2.next
        
        tempHead = head
        
        while(temp1 and temp2):
            if(temp1.val < temp2.val):
                tempHead.next = temp1
                tempHead = tempHead.next
                temp1 = temp1.next
            else:
                tempHead.next = temp2
                tempHead = tempHead.next
                temp2 = temp2.next
        
        if(temp1):
            tempHead.next = temp1
        
        if(temp2):
            tempHead.next = temp2
        
        return head
            